internal/generate/packagemanifest/packagemanifest.go

```diff
@@ -17,37 +17,63 @@ package packagemanifest
 import (
 	"errors"
 	"fmt"
-	"io"
+	"io/ioutil"
 	"path/filepath"
 	"sort"
 
+	"sigs.k8s.io/yaml"
+
 	apimanifests "github.com/operator-framework/api/pkg/manifests"
 	"github.com/operator-framework/api/pkg/validation"
 	log "github.com/sirupsen/logrus"
 
 	genutil "github.com/operator-framework/operator-sdk/internal/generate/internal"
-	"github.com/operator-framework/operator-sdk/internal/generate/packagemanifest/bases"
 )
 
 const (
 	// File extension for all PackageManifests written by Generator.
 	packageManifestFileExt = ".package.yaml"
 )
 
+// generator is an implementation of the Generator interface
+type generator struct{}
+
+// NewGenerator returns a new generator object
+func NewGenerator() generator {
+	return generator{}
+}
+
+// Generator is an interface that specifies the Generate methods
+// to generate and write various package manifests
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . Generator
+type Generator interface {
+	Generate(operatorName, version, outputDir string, opts *Options) error
```

Comment:
This shouldn't be a pointer (matter of preference though):
```suggestion
	Generate(operatorName, version, outputDir string, opts Options) error
```

Reply:
why not? I made it a pointer intentionally so the opts struct (which could potentially grow to be quite large) doesn't have to get shuttled around everywhere

Reply:
I doubt the struct's fields + reference overhead will be > than 1kb, which is not worth fretting over for 2-3 copies. There's also a good chance the compiler will optimize parts of this movement. With pointers, you have to perform a `nil` check, which is an extra jump instruction and more LOC. Perhaps the compiler will optimize this too. I guess my point is that using a pointer makes just as much sense from a performance perspective as pass-by-value, so it comes down to code verbosity and potential nil pointer panics. Since `Options` is a required field, i.e. `opts != nil` will always be true as the code is written now, it makes sense to not use a pointer.